sklearn/neighbors/graph.py

```diff
@@ -176,3 +179,297 @@ def radius_neighbors_graph(X, radius, mode='connectivity', metric='minkowski',
 
     query = _query_include_self(X, include_self)
     return X.radius_neighbors_graph(query, radius, mode)
+
+
+class KNeighborsTransformer(NeighborsBase, KNeighborsMixin,
+                            UnsupervisedMixin, TransformerMixin):
+    """Transform X into a (weighted) graph of k nearest neighbors
+
+    The transformed data is a sparse graph as return by kneighbors_graph.
+
+    Parameters
+    ----------
+    mode : {'distance', 'connectivity'}, optional (default = 'connectivity')
```

Comment:
I think we want the default to be distance. I know this is inconsistent with kneighbors_graph, but I think the default there is a mistake. Users will forget to apply mode='distance' here, and will get bad results.

Reply:
Indeed, I'd be tempted to rename this to `weighted=True/False`, defaulting True, and deprecate `mode` in `kneighbors_graph` to change the default similarly. WDYT?

Reply:
One argument against this change is that mode could also be applied to produce a kernel graph...